### PR TITLE
(feat) Restore ability to collapse and expand the entire form

### PR DESCRIPTION
**Pull request by samuelmale**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR primarily restores the ability to collapse and expand all visible form pages and sections.

**Out of scope**:
1. **Unmounting Issue**: This PR resolves an issue where toggling between expanding and collapsing a page caused its child sections to unmount from the DOM. This was addressed by removing the conditional rendering of sections within the PageRenderer.
2. **UI Enhancements**: Additionally, this PR introduces UI tweaks that visually nest sections within their respective page containers. 

## Screenshots

**Unmounting Issue:**

_(The Encounter Provider and Location fields are remounted)_

**Unmounting Issue Fixed:**

**Other features:**

## Related Issue

N/A

## Other

Note: Carbon doesn't support nesting accordions, which is why I resorted to manipulating the visibility of the sections during page collapse; see:
- https://github.com/carbon-design-system/carbon/issues/7328
- https://github.com/carbon-design-system/carbon/issues/14840

### CODE REVIEW

File: src/components/renderer/page/page.renderer.component.tsx [hunk 1]
```diff
@@ -42,25 +55,51 @@ function PageRenderer({ page }: PageRendererProps) {
...
-                    <SectionRenderer section={section} />
-                  </div>
-                </AccordionItem>
+              {visibleSections.map((section, index) => (
+                <CollapsibleSectionContainer
+                  section={section}
+                  sectionIndex={index}
+                  visibleSections={visibleSections}
+                  isFormExpanded={isFormExpanded}
+                />
```
Comment (denniskigen) [inline src/components/renderer/page/page.renderer.component.tsx:66]:
```suggestion
                <CollapsibleSectionContainer
                	 key={`section-${section.label}`}
                  section={section}
                  sectionIndex={index}
                  visibleSections={visibleSections}
                  isFormExpanded={isFormExpanded}
                />
```

File: src/components/renderer/page/page.renderer.component.tsx [hunk 2]
```diff
@@ -42,25 +55,51 @@ function PageRenderer({ page }: PageRendererProps) {
...
-                <AccordionItem
-                  title={t(section.label)}
-                  open={true}
-                  className={styles.sectionContainer}
-                  key={`section-${section.label}`}>
-                  <div className={styles.formSection}>
-                    <SectionRenderer section={section} />
-                  </div>
-                </AccordionItem>
+              {visibleSections.map((section, index) => (
```
Comment (denniskigen) [inline src/components/renderer/page/page.renderer.component.tsx:60]: Unrelated to your changes, but maybe `visibleSections` could benefit from memoization:

```tsx
const visibleSections = useMemo(() => {
  return page.sections.filter((section) => {
    const hasVisibleQuestions = section.questions.some((question) => !isTrue(question.isHidden));
    return !isTrue(section.isHidden) && hasVisibleQuestions;
  });
}, [page.sections]);
```

File: src/components/renderer/page/page.renderer.component.tsx [hunk 3]
```diff
@@ -42,25 +55,51 @@ function PageRenderer({ page }: PageRendererProps) {
               </span>
             </p>
           </div>
-          {!isCollapsed && (
+          <div className={isCollapsed ? styles.hiddenAccordion : styles.accordionContainer}>
```
Comment (denniskigen) [inline src/components/renderer/page/page.renderer.component.tsx:58]:
```suggestion
          <div className={classNames({
  [styles.hiddenAccordion]: isCollapsed,
  [styles.accordionContainer]: !isCollapsed
})}>
```

File: src/components/renderer/page/page.renderer.component.tsx [hunk 4]
```diff
@@ -42,25 +55,51 @@ function PageRenderer({ page }: PageRendererProps) {
...
+
+  return (
+    <AccordionItem
+      title={t(section.label)}
+      open={isSectionOpen}
+      className={classNames(styles.sectionContainer, {
+        [styles.firstSection]: sectionIndex === 0,
+        [styles.lastSection]: sectionIndex === visibleSections.length - 1,
+      })}
+      key={`section-${section.label}`}>
```
Comment (denniskigen) [inline src/components/renderer/page/page.renderer.component.tsx:97]:
```suggestion
     >
```

**Review by denniskigen — COMMENTED** [review]
Great work, @samuelmale. I've left some comments.

**Review by denniskigen — APPROVED** [review]

**Review by pirupius — APPROVED** [review]
Thanks @samuelmale

**Pull request merged**